unit/src/MatrixToolsTest.C

```diff
@@ -79,11 +74,17 @@ MatrixToolsTest::matrixInversionTest2()
   mat3[7] = m[2][1] = 8.0;
   mat3[8] = m[2][2] = 9.0;
 
-  int error = MatrixTools::inverse(mat3, 3);
-  CPPUNIT_ASSERT(error != 0);
+  try
+  {
+    MatrixTools::inverse(mat3, 3);
+  }
+  catch (const MatrixTools::MatrixInversionException & e) {}
```

Comment:
This test is not well posed now. The test needs to fail if no exception (or the wrong exception) is thrown.